dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -1,59 +1,114 @@
 from datetime import datetime
-
-from dags.analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+from analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+import logging
 
 
 class DiscordTransformRawData(TransformRawDataBase):
     def transform(self, raw_data: list, platform_id: str, period: datetime) -> list[dict]:
         transformed_data = []
         for data in raw_data:
-            interactions = []
-            if data.get("replied"):
-                interactions.append({
-                    "name": "reply",
-                    "users_engaged_id": [data.get("replied")],
-                    "type": "receiver"
-                })
-
-            if data.get("user_mentions"):
-                interactions.append({
-                    "name": "mention",
-                    "users_engaged_id": data.get("user_mentions"),
-                    "type": "emitter"
-                })
-
-            if data.get("reactions"):
-                for reaction in data.get("reactions"):
-                    parts = reaction.split(", ")
-                    if len(parts) > 2:
+            try:
+                if "author" not in data or not data["author"]:
+                    raise ValueError("Missing 'author' in raw data")
+
+                interactions = []
+
+                if data.get("replied_user"):
+                    interactions.append({
+                        "name": "reply",
+                        "users_engaged_id": [data["replied_user"]],
+                        "type": "emitter"
+                    })
+                    # Create another document for the receiver
+                    receiver_interaction = {
+                        "author_id": data["replied_user"],
+                        "date": period,
+                        "source_id": data["messageId"],
+                        "metadata": {
+                            "thread_id": data["threadId"],
+                            "channel_id": data["channelId"],
+                            "bot_activity": data["isGeneratedByWebhook"] or data["botActivity"],
```

Comment:
the bot_activity would be identified via two conditions
1. the `isGeneratedByWebhook` field which is already written.
2. If the user was a bot identified on members collection. The members collection for discord is called `guildmembers`

the guildmembers collection structure is in the https://rndadocs.notion.site/Analyzer-Discord-Pipeline-5009512d82a243deb35b3388e17b29c5#:~:text=guildmembers%20collection%20current%20data%20structure 

You could supply this function with guildmembers data (just the `discordId` and `isBot` field). Also, Please feel free to share thoughts, if you have other opinions on this.